.github/workflows/doc-lint-test.yaml

```diff
@@ -0,0 +1,459 @@
+      - name: Configure Git safe directory
+        run: git config --global --add safe.directory $PWD
+
+      - name: Install geoips documentation packages
+        run: pip install -e .[doc]
+
+      - name: Run build html
+        run: |
+          echo "Build html docs"
+          # Call build_docs.sh from the geoips default branch
+          # Pass in
+          #   the path to the current plugin repo,
+          #   the name of the current repo,
+          #   "html_only" to indicate producing only html output, and
+          #   the path to the docs template directory in the geoips default branch.
+          $BASE_GEOIPS_PATH/docs/build_docs.sh . $CURR_REPO html_only ./docs
+          ret=$?
+          if [[ "${ret##*:}" != *"0"* ]]; then
+            echo "::error::Building html docs ${ret##*:}"
+            exit 1
+          fi
+
+# This job does not currently work because latex is not installed in the image. I think
+# that to get this job working would only require installing texlive-base but other
+# texlive packages might also be required.
+#
+# Additionally, this job is extremely slow. Building the PDF documentation is time
+# consuming and should probably only be run on releases. Building the HTML documentation
+# should be enough to catch any problems that crop up.
+#
+#   build_sphinx_pdf:
+#     if: ${{ needs.set_variables.outputs.CI_DISABLE_BUILD_SPHINX_PDF == 'false' }}
+#     runs-on: ${{ vars.RUNNER }}
+#     needs: [build_doclinttest_image, set_variables]
+#     container:
+#       image: ${{ needs.set_variables.outputs.DOCLINTTEST_IMAGE }}
+#       credentials:
+#         username: ${{ secrets.DOCKER_REGISTRY_USER }}
+#         password: ${{ secrets.DOCKER_REGISTRY_TOKEN }}
+#     env:
+#       CURR_REPO: ${{ github.event.repository.name }}
+#     steps:
+#       - name: Checkout code
+#         uses: actions/checkout@v4
+#
+#       - name: Configure Git safe directory
+#         run: git config --global --add safe.directory $PWD
+#
+#       - name: Install geoips documentation packages
+#         run: pip install -e .[doc]
+#
+#       - name: Run build pdf
+#         run: |
+#           echo "Build pdf docs"
+#           # Call build_docs.sh from the geoips default branch
+#           # Pass in
+#           #   the path to the current plugin repo,
+#           #   the name of the current repo,
+#           #   "pdf_only" to indicate producing only pdf output, and
+#           #   the path to the docs template directory in the geoips default branch.
+#           ./docs/build_docs.sh . $CURR_REPO pdf_only ./docs
+#           ret=$?
+#           if [[ "${ret##*:}" != *"0"* ]]; then
+#             echo "::error::Building pdf docs ${ret##*:}"
+#             exit 1
+#           fi
+
+#################################################
+# Linting
+#################################################
+  lint_bandit:
+    if: >
+      ${{ env.CI_LINT_BANDIT == 'true' }} &&
+      (needs.build_doclinttest_image.result == 'success' ||
```

Comment:
Right. This took a LONG time to get right. 

If `build_doclinttest_image` either fails or is skipped, `lint_bandit` would not run by default. Adding `always()` to the beginning of the conditional tells github to continue even if the previous job failed or was skipped. Then the tests of `result` ensure that we only run `lint_bandit`, and other similar jobs, if `result` contains "success" or "skipped" but not if it contains "failed" or some other value.

I just realized something that could simplify this. I could add an intermediate job that looks like this:
```
run_dependent_jobs_on_success_or_skip:
  if: always()
  run: |
    prev_result = ${{ needs.build_doclinttest.result }}
    if [[ ${prev_result} == 'success' || ${prev_result} == 'skipped' ]]; then
        echo "Parent job status '${prev_result}'. Will run dependent jobs."
        exit 0
    else
        echo "Parent job status '${prev_result}'. Will not run dependent jobs."
        exit 1
    fi
```

I think I'll give that a shot before I'm completely done with this since it will simplify things significantly.

Reply:
You can open an issue for this!